azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -136,12 +133,17 @@ protected void deployArtifacts(WebAppConfiguration webAppConfiguration)
             final boolean isV1Schema = SchemaVersion
                     .fromString(this.getSchemaVersion()) == SchemaVersion.V1;
             // copy resources according to maven filter if the os is not docker
-            if (webAppConfiguration.getOs() != OperatingSystemEnum.Docker && isV1Schema && this.getDeploymentType() != DeploymentType.JAR) {
-                final List<Resource> effectiveResources = isV1Schema ? resources
+            if (webAppConfiguration.getOs() != OperatingSystemEnum.Docker) {
+                final List<Resource> finalResources = isV1Schema ? resources
                                 : (this.deployment == null ? null : this.deployment.getResources());
-                if (effectiveResources != null && !effectiveResources.isEmpty()) {
-                    copyArtifactsToStagingDirectory(isV1Schema ? effectiveResources :
-                        effectiveResources.stream().filter(resource -> !ArtifactHandlerImplV2.isExternalResource(resource)).collect(Collectors.toList()));
+                if (!isV1Schema) {
+                    final Map<Boolean, List<Resource>> resourceMap = finalResources.stream()
+                            .collect(Collectors.partitioningBy(ArtifactHandlerImplV2::isExternalResource));
+                    this.setExternalResources(resourceMap.get(true));
+                    this.deployment.setResources(resourceMap.get(false));
+                }
+                if (finalResources != null && !finalResources.isEmpty()) {
```

Comment:
Shall we check resources(wwwroot only) instead of finalResources as may be all resources are external